crates/extensions/c8y_mapper_ext/src/actor.rs

```diff
@@ -50,6 +51,8 @@ pub struct C8yMapperActor {
     messages: SimpleMessageBox<C8yMapperInput, C8yMapperOutput>,
     mqtt_publisher: LoggingSender<MqttMessage>,
     timer_sender: LoggingSender<SyncStart>,
+    ops_dir: PathBuf,
+    tedge_agent_status: bool,
```

Comment:
I understand that you added this flag to prevent the mapper from repeatedly sending the software list request to the agent every time its status changes. But, may be it's a good thing. What if the agent restarted with a new set of sm-plugins loaded. In that case, it might be better to send the software list request to the agent every time it says it's "up".

Reply:
You mean send the `software list` request only when the status is `up`?

Reply:
Yeah, send the software list request every time we receive an `up` status. Then, we don't even need this flag.

Reply:
But the plan was to send if `tedge-agent` is up at least once. i.e up/down any message on `health` topic.